thirdparty/github.go

```diff
@@ -215,44 +256,34 @@ func githubShouldRetry(caller string, config retryConfig) utility.HTTPRetryFunct
 	}
 }
 
-// getGithubClient returns an *http.Client that will retry according to supplied retryConfig.
-// Also creates a span tracking the lifespan of the client.
-// Defer the returned function to release the client to the pool and close the span.
-func getGithubClient(ctx context.Context, token, caller string, config retryConfig, attributes []attribute.KeyValue) (context.Context, *http.Client, func()) {
+func getGithubClient(token, caller string, config retryConfig) *github.Client {
```

Comment:
nit: do we maybe want to add a comment saying that the client will use caching?

Reply:
done